File: csharp/isbn-verifier/IsbnVerifier.cs
using System;
using System.Collections.Generic;
using System.Linq;
public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        if (number == "") return false;
        bool isXCheck = number[number.Length-1] == 'X' ? true : false;
        int total = 0;
        string cleanedString = new string (number.Where(s => char.IsDigit(s)).ToArray());
        if (!(cleanedString.Length == 10 && !isXCheck) && !(cleanedString.Length == 9 && isXCheck))
            return false;
        for (int i=0; i < 9; i++)
            total += int.Parse(cleanedString[i].ToString()) * (10-i);
        if (isXCheck) total += 10;
        else total += int.Parse(cleanedString[cleanedString.Length-1].ToString());
        if (total%11 == 0) 
            return true;
        else
            return false;
    }
}

File: java/isbn-verifier/src/main/java/IsbnVerifier.java
class IsbnVerifier {

    boolean isValid(String stringToVerify) {
        String tmpStr = stringToVerify.replace("-", "");
        if (tmpStr.length() != 10)
            return false;
        Boolean isX = tmpStr.charAt(tmpStr.length()-1) == 'X';
        int retVal = 0;
        tmpStr = tmpStr.replace("X", "");
        for (int i=0, j=10; i < tmpStr.length(); i++, j--) {
            if (!Character.isDigit(tmpStr.charAt(i)))
                return false;
            retVal += Integer.parseInt(Character.toString(tmpStr.charAt(i))) * j;
        }
        if (isX)
            retVal += 10;
        return retVal%11 == 0;
    }

}
